scripts/download_library.py

```
@@ -0,0 +1,341 @@
+                license_file = folder / "LICENSE"
+                fd.extract("LICENSE", tmp_dir)
+                shutil.move(os.fspath(tmp_dir / "LICENSE"), os.fspath(license_file))
+
+                lib_file = folder / "libmediainfo.so.0"
+                fd.extract("lib/libmediainfo.so.0.0.0", tmp_dir)
+                shutil.move(os.fspath(tmp_dir / "lib/libmediainfo.so.0.0.0"), os.fspath(lib_file))
+
+        # MacOS (darwin)
+        elif compressed_file.endswith(".tar.bz2") and self.platform == "darwin":
+            with tarfile.open(file) as fd:
+                kwargs: dict[str, Any] = {}
+                if sys.version_info >= (3, 12):
+                    kwargs = {"filter": "data"}
+
+                license_file = folder / "License.html"
+                fd.extract("MediaInfoLib/License.html", tmp_dir, **kwargs)
+                shutil.move(
+                    os.fspath(tmp_dir / "MediaInfoLib/License.html"),
+                    os.fspath(license_file),
+                )
+
+                lib_file = folder / "libmediainfo.0.dylib"
+                fd.extract("MediaInfoLib/libmediainfo.0.dylib", tmp_dir, **kwargs)
+                shutil.move(
+                    os.fspath(tmp_dir / "MediaInfoLib/libmediainfo.0.dylib"),
+                    os.fspath(lib_file),
+                )
+
+        # Windows (win32)
+        elif compressed_file.endswith(".zip") and self.platform == "win32":
+            with ZipFile(file) as fd:
+                license_file = folder / "License.html"
+                fd.extract("Developers/License.html", tmp_dir)
+                shutil.move(os.fspath(tmp_dir / "Developers/License.html"), os.fspath(license_file))
+
+                lib_file = folder / "MediaInfo.dll"
+                fd.extract("MediaInfo.dll", tmp_dir)
+                shutil.move(os.fspath(tmp_dir / "MediaInfo.dll"), os.fspath(lib_file))
+
+        files = {}
+        if license_file is not None and license_file.is_file():
```

Should we raise an error if license_file is defined but the file doesn't exist?

---

Probably an error would be raised by `fd.extract` but it does not hurt to add it here too.

---

You're right, I missed this part. I tested it with zipfile and tarfile and both raise a `KeyError`. Let's keep it this way.